src/main/scala/firrtl/transforms/ConstantPropagation.scala

```diff
@@ -99,14 +108,22 @@ class ConstantPropagation extends Transform with ResolvedAnnotationPaths {
 
   override val annotationClasses: Traversable[Class[_]] = Seq(classOf[DontTouchAnnotation])
 
-  trait FoldCommutativeOp {
+  sealed trait SimplifyBinaryOp {
+    def matchingArgsValue(e: Expression, arg: Expression): Expression
```

Comment:
```suggestion
    def matchingArgsValue(e: DoPrim, arg: Expression): Expression
```
We know that `e` is always a `DoPrim` right? I think we should use the more precise type if we can.

Reply:
Yes. I think I made it `Expression` just to not differ from `FoldCommutativeOp`, which I couldn't fix for backport reasons, but that's a bad reason.